### PR TITLE
Feat: Store invoice numbers and totals when batch is sent

### PR DESCRIPTION
**Pull request by DGajewska**
WATER-2976

* Adds invoice_number to water.billing_invoices
* Adds invoiceNumber to Invoice model
* Update CM decorators to use cm transaction details, if availble
* Refactors CM decorators
* Persists invoice number and totals data

### CODE REVIEW

**Review by james-siteclick — CHANGES_REQUESTED** [review]
Looks really good generally, couple of minor questions

File: src/lib/connectors/repos/billing-invoices.js [hunk 1]
```diff
@@ -57,11 +57,13 @@ const deleteByBatchId = async (batchId, isDeletionRequired = true) => BillingInv
   .where({ billing_batch_id: batchId })
   .destroy({ require: isDeletionRequired });
 
-exports.deleteEmptyByBatchId = deleteEmptyByBatchId;
-exports.findOne = findOne;
+const update = async (billingInvoiceId, changes) => BillingInvoice
```
[the remark this answers is not in the document]
Comment (DGajewska) [reply src/lib/connectors/repos/billing-invoices.js:60]: Yes, thank you. Updated

File: src/modules/billing/mappers/charge-module-decorators.js [hunk 2]
```diff
@@ -7,18 +7,68 @@ const Batch = require('../../../lib/models/batch');
...
+};
+
+const getCmFinancialYearForTransaction = transaction => getFinancialYear(transaction.periodStart) - 1;
+
+const groupTransactionsByFinancialYear = transactions =>
+  groupBy(transactions, getCmFinancialYearForTransaction);
+
+const groupTransactionsByCustomerAndFinancialYear = cmTransactions => {
+  const groupedByCustomer = groupBy(cmTransactions, 'customerReference');
+  return mapValues(groupedByCustomer, groupTransactionsByFinancialYear);
```
[the remark this answers is not in the document]
Comment (DGajewska) [reply src/modules/billing/mappers/charge-module-decorators.js:38]: I have simplified this based on your suggestions. Thank you

File: src/modules/billing/mappers/charge-module-decorators.js [hunk 3]
```diff
@@ -7,18 +7,68 @@ const Batch = require('../../../lib/models/batch');
...
 
+  const transactionsByCustomerAndFinYear = groupTransactionsByCustomerAndFinancialYear(cmTransactions.data.transactions);
+  return mapCmTransactionsToCustomers(cmResponse.billRun.customers, transactionsByCustomerAndFinYear);
+};
+
+/**
+ * Find invoice number, or return null if none present
+ */
+const findInvoiceNumber = customerFinYearSummary => {
+  const transactionWithInvoiceNumber = customerFinYearSummary.transactions.find(transaction => !!transaction.transactionReference);
```
[the remark this answers is not in the document]
Comment (DGajewska) [reply src/modules/billing/mappers/charge-module-decorators.js:67]: Yes, this is a good point. I've now included this :)

**Review by james-siteclick — APPROVED** [review]

**Pull request merged**